### PR TITLE
Remove `k_` prefix from seed variables

### LINKED ISSUE #435: Remove leading "`k_`" in variable names
This was meant to denote that something is an XOF seed, but not that we're making good use of type hints, this doesn't seem necessary anymore.

### PR DESCRIPTION
**Pull request by cjpatton**
Closes #435.

Also, add a `_seed` suffix where the new variable name is confusing: `k_prove` -> `prove_seed` in Prio3; and `k_shard` -> `shard_seed` in Poplar1.

### CODE REVIEW

File: draft-irtf-cfrg-vdaf.md [hunk 1]
```diff
@@ -3075,18 +3075,18 @@ def expand_input_share(
...
+        proofs_share = self.helper_proofs_share(ctx, agg_id, share)
     else:
         assert len(input_share) == 3
-        (meas_share, proofs_share, k_blind) = input_share
-    return (meas_share, proofs_share, k_blind)
+        (meas_share, proofs_share, blind) = input_share
+    return (meas_share, proofs_share, blind)
 
-def prove_rands(self, ctx: bytes, k_prove: bytes) -> list[F]:
+def prove_rands(self, ctx: bytes, prove: bytes) -> list[F]:
```
Comment (divergentdave) [inline draft-irtf-cfrg-vdaf.md:3086]: This should be renamed as well.
```suggestion
def prove_rands(self, ctx: bytes, prove_seed: bytes) -> list[F]:
```

File: poc/vdaf_poc/vdaf_prio3.py [hunk 2]
```diff
@@ -440,18 +440,18 @@ def expand_input_share(
...
+            proofs_share = self.helper_proofs_share(ctx, agg_id, share)
         else:
             assert len(input_share) == 3
-            (meas_share, proofs_share, k_blind) = input_share
-        return (meas_share, proofs_share, k_blind)
+            (meas_share, proofs_share, blind) = input_share
+        return (meas_share, proofs_share, blind)
 
-    def prove_rands(self, ctx: bytes, k_prove: bytes) -> list[F]:
+    def prove_rands(self, ctx: bytes, prove: bytes) -> list[F]:
```
Comment (divergentdave) [inline poc/vdaf_poc/vdaf_prio3.py:451]: Same here
```suggestion
    def prove_rands(self, ctx: bytes, prove_seed: bytes) -> list[F]:
```

File: draft-irtf-cfrg-vdaf.md [hunk 3]
```diff
@@ -2755,36 +2755,36 @@ def shard_with_joint_rand(
...
         for i in range(0, (self.SHARES - 1) * 2, 2)
     ]
-    k_helper_blinds = [
-        k_helper_seeds[i]
+    helper_blinds = [
+        helper_seeds[i]
         for i in range(1, (self.SHARES - 1) * 2, 2)
     ]
-    (k_leader_blind, k_prove), seeds = front(2, seeds)
+    (leader_blind, prove), seeds = front(2, seeds)
```
[the remark this answers is not in the document]
Comment (cjpatton) [reply draft-irtf-cfrg-vdaf.md]: Done.

File: draft-irtf-cfrg-vdaf.md [hunk 4]
```diff
@@ -2937,12 +2937,12 @@ def prep_next(
     prep_state: Prio3PrepState[F],
     prep_msg: Optional[bytes]
 ) -> tuple[Prio3PrepState[F], Prio3PrepShare[F]] | list[F]:
-    k_joint_rand = prep_msg
-    (out_share, k_corrected_joint_rand) = prep_state
+    joint_rand = prep_msg
```
[the remark this answers is not in the document]
Comment (cjpatton) [reply draft-irtf-cfrg-vdaf.md]: Agreed, done.

File: draft-irtf-cfrg-vdaf.md (hunk 1 above)
Comment (cjpatton) [reply draft-irtf-cfrg-vdaf.md]: Done.

**Review by divergentdave — APPROVED** [review]

**Pull request merged**